internal/resource/http.go

```diff
@@ -181,15 +186,23 @@ func (f *Fetcher) newHttpClient() {
 				PreferGo: true,
 			},
 		}).Dial,
+		TLSClientConfig:     &tlsConfig,
 		TLSHandshakeTimeout: 10 * time.Second,
 	}
+	client := http.Client{
+		Transport: &transport,
+	}
+	return &client, &transport
+}
+
+// newHttpClient populates the fetcher with the default HTTP client.
+func (f *Fetcher) newHttpClient() {
+	defaultClient, defaultTransport := defaultHTTPClient()
 	f.client = &HttpClient{
-		client: &http.Client{
-			Transport: transport,
-		},
+		client:    defaultClient,
 		logger:    f.Logger,
 		timeout:   time.Duration(defaultHttpTotalTimeout) * time.Second,
-		transport: transport,
+		transport: defaultTransport,
```

Comment:
Couldn't this grab the transport from the client so the second return value from `defaultHTTPClient()` is unnecessary?